tests/contrib/sensors/test_sagemaker_base_sensor.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+from airflow import configuration
+from airflow.contrib.sensors.sagemaker_base_sensor import SageMakerBaseSensor
+from airflow.exceptions import AirflowException
+
+
+class TestSagemakerBaseSensor(unittest.TestCase):
+    def setUp(self):
+        configuration.load_test_config()
+
+    def test_subclasses_succeed_when_response_is_good(self):
+        class SageMakerBaseSensorSubclass(SageMakerBaseSensor):
+            NON_TERMINAL_STATES = ['PENDING', 'RUNNING', 'CONTINUE']
+            FAILED_STATE = ['FAILED']
+
+            def get_sagemaker_response(self):
+                return {
+                    'SomeKey': {'State': 'COMPLETED'},
+                    'ResponseMetadata': {'HTTPStatusCode': 200}
+                }
+
+            def state_from_response(self, response):
+                return response['SomeKey']['State']
+
+        operator = SageMakerBaseSensorSubclass(
+            task_id='test_task',
+            poke_interval=2,
+            aws_conn_id='aws_test'
+        )
+
+        operator.execute(None)
+
+    def test_poke_returns_false_when_state_is_a_non_terminal_state(self):
+        class SageMakerBaseSensorSubclass(SageMakerBaseSensor):
+            NON_TERMINAL_STATES = ['PENDING', 'RUNNING', 'CONTINUE']
+            FAILED_STATE = ['FAILED']
+
+            def get_sagemaker_response(self):
+                return {
+                    'SomeKey': {'State': 'PENDING'},
+                    'ResponseMetadata': {'HTTPStatusCode': 200}
+                }
+
+            def state_from_response(self, response):
+                return response['SomeKey']['State']
+
+        operator = SageMakerBaseSensorSubclass(
+            task_id='test_task',
+            poke_interval=2,
+            aws_conn_id='aws_test'
+        )
+
+        self.assertEqual(operator.poke(None), False)
+
+    def test_poke_returns_false_when_method_not_implemented(self):
```

Comment:
I think it doesn't return false. Should be test_poke_raise_exception_......

Reply:
changed the method name